src/MetadataManager/EventManager.ts

```diff
@@ -0,0 +1,253 @@
+            await Metadata.disableMetadataUnderFolder(uri);
+          } else {
+            // case 1. [File]+undefined | deactive
+            await Metadata.disableMetadata(uri);
+          }
+        }
+      }),
+      provider.fileWatcher.onDidCreate(async uri => {
+        provider.refresh('Create'); // test code
+        console.log('onDidCreate  '+uri.fsPath);
+        MetadataEventManager.createUri=uri;
+
+        const instance= await PathToHash.getInstance();
+        if(fs.statSync(uri.fsPath).isDirectory()){
+          // case 1. [Dir]  Copy with files > Serch all the file in the Dir
+          await provider.createDirEvent(uri);
+        }
+        else if(Metadata.isValidFile(uri)){
+          if(instance.getPathToHash(uri)&&workspaceRoot){
+            //case 2. [File] Contents change event in Ubuntu terminal (refer to pathToHash)
+            await provider.changeEvent(uri);
+          }
+        else{
+          // case 3. [File] File generation event
+          await provider.createFileEvent(uri);
+        }
+        }
+        MetadataEventManager.createUri=undefined;       
+
+      }),
+    ];
+
+    registrations.forEach(disposable => context.subscriptions.push(disposable));
+  }
+
+  constructor() {
+  }
+
+  refresh(message: string): void {
+    vscode.window.showInformationMessage(message);
+  }
+
+
+  async changeEvent(uri:vscode.Uri): Promise<void> {
+    if(!Metadata.isValidFile(uri)) {return;}
+    // case 1. [File] Contents change event
+    const relativePath = vscode.workspace.asRelativePath(uri);
+
+    //(1) get beforehash and set afterhash
+    const beforehash: string = await Metadata.getFileHash(uri);
+    const buffer = Buffer.from(await vscode.workspace.fs.readFile(uri)).toString();
+    const afterhash: string = crypto.createHash('sha256').update(buffer).digest('hex');
+
+    //(2) deactivate changed hash object
+    let metadata: any = await Metadata.getMetadata(beforehash);
+    if(Object.keys(metadata).length !== 0 && metadata[relativePath]) {
+        metadata[relativePath]["isDeleted"] = true;
+        await Metadata.setMetadata(beforehash, metadata);
+    }
+
+    //(3) change pathToHash
+    const instance = await PathToHash.getInstance();
+    await instance.addPath(uri);
+
+    metadata = await Metadata.getMetadata(afterhash);
+
+    //(4) create new hash object
+    if (Object.keys(metadata).length === 0) {
+      await Metadata.setMetadata(afterhash, {});
+    }
+    
+    const filename: any = relativePath.split('/').pop();
+    const stats: any = await MetadataEventManager.getStats(uri);
+
+    metadata[relativePath] = {};
+    metadata[relativePath]["name"] = filename;
+
+    metadata[relativePath]["fileExtension"] = filename.split(".").at(-1);
+    metadata[relativePath]["createTime"] = stats.birthtime;
+    metadata[relativePath]["modifiedTime"] = stats.mtime;
+    metadata[relativePath]["isDeleted"] = false;
+
+    await Metadata.setMetadata(afterhash, metadata);    
```

Comment:
As of our offline conversation, you can extract this as a `Metadata`'s function or so.